### PR TITLE
Feature following

### PR DESCRIPTION
**Pull request by andela-ijubril**

### CODE REVIEW

**Comment by andela-ijubril** [comment]
My bad, i'm sorry i did not delete it before sending

On Tue, Nov 17, 2015 at 10:54 PM, Chidiebere Nnadi <notifications@github.com

**Comment by andela-ijubril** [comment]
I have also removed the line of code

File: codango/userprofile/templates/userprofile/partials/follow_list.html [hunk 1]
```diff
@@ -0,0 +1,51 @@
+{% load cloudinary %}
+
+<div class="content-block">
+
+    <div class="panel panel-default">
+        <div class="panel-heading" id="followers-heading">
+            {% if request.path == '/user/'|add:profile.user.username|add:'/following' %}
```
Comment (andela-cnnadi) [inline codango/userprofile/templates/userprofile/partials/follow_list.html]: **Code Quality: Nitpick**

Why don't you just save the direction variable within the context and use that within this condition? I think that should be much more straightforward.

File: codango/userprofile/tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+__author__ = 'masterp'
```
Comment (andela-cnnadi) [inline codango/userprofile/tests/__init__.py]: **Code Quality: Critical**

**PyCharm** has done it again :wink:

File: codango/userprofile/views.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        userprofile.save()
+
+        follower_user_profile = UserProfile.objects.get(user_id=following_id.id)
+        follower_user_profile.followers += 1
+        follower_user_profile.save()
+
+        return HttpResponse(status=200)
+
+
+class FollowingView(LoginRequiredMixin, TemplateView):
```
Comment (andela-cnnadi) [inline codango/userprofile/views.py]: **Code Quality: Nitpick**

Not sure you still need this class and also the `FollowersView`.

File: codango/userprofile/tests/test_views.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    def setUp(self):
+        self.client = Client()
+        self.user1 = User.objects.create_user(username='golden', password='abiodun')
+        self.user2 = User.objects.create_user(username='jubril', password='issa')
+        self.user1.save()
+        self.user2.save()
+        self.login = self.client.login(username='golden', password='abiodun')
+
+
+    def test_a_logged_in_user_can_follow_a_registered_user(self):
```
Comment (andela-cnnadi) [inline codango/userprofile/tests/test_views.py]: **Code Quality: Nitpick**

You shouldn't have empty tests. Not recommended.

File: codango/userprofile/templates/userprofile/partials/follow_list.html [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                                <span><a href="/user/{{ following.followed }}"
+                                         id="follower-name"> {{ following.followed }}</a></span><br/>
+                                <div class="clearfix"></div>
+                                </div>
+                            {% else %}
+                                {% cloudinary "vqr7n59zfxyeybttleug.gif" width=100 height=150 crop="fill" %}
+                            {% endif %}
+                    {% endfor %}
+                </div>
+                {% endif %}
```
Comment (andela-cnnadi) [inline codango/userprofile/templates/userprofile/partials/follow_list.html]: **Code Quality: Nitpick**

This line should be indented backwards a single step.

File: codango/userprofile/templates/userprofile/partials/follow_list.html [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+                                </div>
+                            {% else %}
+                                {% cloudinary "vqr7n59zfxyeybttleug.gif" width=100 height=150 crop="fill" %}
+                            {% endif %}
+                    {% endfor %}
+                </div>
+
+            {% endif %}
+
+</div>
```
Comment (andela-cnnadi) [inline codango/userprofile/templates/userprofile/partials/follow_list.html]: **Code Quality: Nitpick**

Indentation issue here too.

**Pull request merged**